src/session_handle.cpp

```diff
@@ -760,12 +760,14 @@ namespace {
 
 	unsigned short session_handle::listen_port() const
 	{
-		return sync_call_ret<unsigned short>(&session_impl::listen_port);
+		return sync_call_ret<unsigned short, unsigned short(session_impl::*)() const>
```

Comment:
We can certainly deprecate this. The only potential issue is that it would leave the `listen_succeeded_alert` as the only way for the client to find out what port(s) it is listening on.